### PR TITLE
Jeannie's API muncher

### PR DESCRIPTION
**Pull request by BJHunnicutt**

### CODE REVIEW

File: app/controllers/homepages_controller.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  end
+
+  def show
+    @recipe = EdamamApiWrapper.get_recipe(params[:id])
+    @servings = @recipe.servings
+    @calories = @recipe.calories/@servings
+    @nutrition_info = @recipe.nutrition_info
+  end
+
+  def get_total(recipes)
```
Comment (CheezItMan) [inline app/controllers/homepages_controller.rb:25]: `get_total` should be private if it's an internal helper method.

File: lib/edamam_api_wrapper.rb [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                      servings: recipe["recipe"]["yield"],
+                      calories: recipe["recipe"]["calories"],
+                      source: recipe["recipe"]["source"],
+                      count: data["count"] #Only getting this for list_recipes
+                      })
+        recipes << wrapper
+      end
+     return recipes
+    else
+     return nil
```
Comment (CheezItMan) [inline lib/edamam_api_wrapper.rb:45]: Good handling of getting an unexpected response.

File: lib/edamam_api_wrapper.rb [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  end
+
+  def self.total_recipes(search_term)
+    if !search_term.nil?
+      search_term = URI.encode(search_term) # add %20 to spaces
+    end
+
+    url = BASE_URL + "search?app_id=#{EDAMAM_ID}" + "&app_key=#{EDAMAM_KEY}" + "&q=#{search_term}"
+
+    data = HTTParty.get(url)
```
Comment (CheezItMan) [inline lib/edamam_api_wrapper.rb:79]: Just to confirm when you use this you'll end up making 2 API calls, that is a bit of a performance penalty.  You might think down-the-road about what you could do to optimize the app.

File: lib/recipe.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
+require 'awesome_print'
+
+class Recipe
+  attr_reader :name, :id, :url, :health_labels, :diet_labels, :cautions, :instructions, :ingredients, :image, :edamam_url, :servings, :calories, :source, :nutrition_info, :count
+
+  def initialize(name, id, options = {})
+
+    if name == nil || id == nil || name == "" || id == ""
+      raise ArgumentError
+    end
```
Comment (CheezItMan) [inline lib/recipe.rb:10]: Good work requiring necessary fields.

File: test/controllers/homepages_controller_test.rb [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    VCR.use_cassette("total_error_NEXT_search_index") do
+      get :index, {search_term: "hamburger pasta basil", next: 0} #This search returns 12 recepes, but the API count is 14, and my trick of subtracting 1 for every 10 search results still leaves the total at 13...
+      assert_response :success
+      assert_not @controller.params[:back]
+      assert_equal @controller.params[:next], "0"
+      assert_equal @controller.params[:from], 12
+      assert_equal @controller.params[:to], 24
+      assert_not_equal @controller.instance_variable_get(:@total), 14 #14 is what the api says it is
+      assert_equal @controller.instance_variable_get(:@total), 12 #12 is what the number of actual recipes returned (only knows this when the last set of recepes is delivered)
+
```
Comment (CheezItMan) [inline test/controllers/homepages_controller_test.rb:71]: Good work using `ap` to figure out what you can do to test the instance variables.

File: test/controllers/homepages_controller_test.rb [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+      # ap @controller.params ## OMG!!!!!!!!!!!!!!!!!!
+      # ap @response
+      # ap @response.status   # Exact status code!
+      # ap @controller.instance_variables  # SO GOOD!!!!!!!!!
+      # ap @controller.instance_variable_get(:@total)
+    end
+  end
+
+end
```
Comment (CheezItMan) [inline test/controllers/homepages_controller_test.rb:83]: You should also probably test the negative case and make sure that you handle cases where the search request or the show page is invalid.

**Review by CheezItMan — COMMENTED** [review]
@BJHunnicutt Nice work on this project.  Very cleanly written and it executes well.    I left a few notes in your code, but overall, very impressive.
